### ISSUE TITLE
Update snap to core20

### ISSUE THREAD

**Issue by pachulo** [nextcloud-snap/nextcloud-snap#2011]
Well, we still have 14 months of support for `core18`, but I guess it's time to start the discussion about this, or maybe not yet @kyrofa ?

On the other hand, I don't see any particular benefit from upgrading...anyone knows of any?

**Comment by kyrofa** [comment]
Definitely worth starting the discussion, thanks @pachulo! I also don't know offhand of any benefits to upgrading. That along with the fact that snapcraft and launchpad tend to

1. Lag pretty significantly in supporting new bases, and
2. Hide breakage behind the use of a particular base

I prefer to wait until nearly the last possible moment to do that kind of upgrade.

**Comment by kyrofa** [comment]
Sadly no. Snapcraft hides (or at least tries to hide) all of its breaking changes behind base changes. It's always a tremendously painful experience. We need to figure out what to do about our plugins-- I think they've gotten rid of that functionality altogether, although we might still be able to use some legacy APIs.

**Comment by airlink75** [comment]
Hello,
Any news about this ?
Many thanks.

**Comment by airlink75** [comment]
> After all, Ubuntu 18.04 and with it `core18` went end of standard support two weeks ago.

Yeah obviously 😅. I was asking about news for a core20 update cause for me it's a security issue to stay with core18.

**Comment by kyrofa** [comment]
> it's a security issue to stay with core18

This isn't actually true, folks. Ubuntu Core 18, built on top of the core18 snap, is supported for ten years. I'd still like to get onto core20, but it's not a burning need.

**Comment by airlink75** [comment]
For me it is, cause i'm running nextcloud-snap inside a LXC container.
Currently I can't update my LXC container further than ubuntu 18.04 cause of nextcloud-snap staying in core18.
If I upgrade my LXC container I am facing issues with apparmor cause the core18 kernel doesn't support it.

So definitely for me it's an issue cause I need to keep my Ubuntu 18.04 container. I see 3 options :
1. Switch to Nextcloud standard version (but i need to update each new version manually => I don't want to do this. 
2. Don't use LXC => Means I need to migrate my Nextcloud database and that I lose the flexibility of my container. Not ideal.
3. Wait for an update to core20.

**Comment by kyrofa** [comment]
> Currently I can't update my LXC container further than ubuntu 18.04 cause of nextcloud-snap staying in core18.
If I upgrade my LXC container I am facing issues with apparmor cause the core18 kernel doesn't support it.

I'm not following. Even the `core` snap based on Ubuntu 16.04 is still installable. Later versions of Ubuntu should happily be able to install the core18 snap without issue. The "core18 kernel" is not a thing here-- you shouldn't have a kernel snap on a class Ubuntu install.

**Comment by kyrofa** [comment]
> I don't know any reason, why this should be different for core18 and Canonical should give away the support for free (ignoring the special cases, where you are entitled for ESM for free).

Well, think about it. Ubuntu Core is supported for ten years. What is Ubuntu Core? It's a Linux distribution made up of a few specific snaps:

- A gadget
- A kernel
- A base (e.g. core18 on Ubuntu Core 18)
- Snapd
- Application snaps

The base snap, core18, is the same core18 we're building on. You don't need ESM to use Ubuntu Core, but it's supported for ten years. If that's true, they need to continue releasing core18 updates to those existing Ubuntu Core devices. Ergo, they need to continue releasing core18 updates to everyone. That includes us. There's no other way to do it, as far as I can tell, without somehow authenticating every Ubuntu Core device and only providing updates to it. They didn't have that ability (or desire) last I checked.

**Comment by kyrofa** [comment]
As an example, note that the `core` snap, again based on 16.04, was updated recently:

```
$ snap info core
name:      core
<snip>
refresh-date: 2023-03-21
```

**Comment by kyrofa** [comment]
> I find Canonicals Documentation on snaps insufficient, highly confusing

Oh yes, no debate there.

**Comment by pachulo** [comment]
See #2908

**Comment by VIXI0** [comment]
**Trivial comment ahead** 
I didn't know the big picture of issue above, I hope you find the proper solution, keep doing the great job you being doing!!

**Comment by pachulo** [comment]
Now that we got rid of local plugins, it's time to try to upgrade the core used by the snap.
See here for what needs to be changed: https://documentation.ubuntu.com/snapcraft/stable/how-to/change-bases/change-from-core18-to-core20/

**Comment by scubamuc** [comment]
Hey @airlink75 my LXC's are all 100% up to date! there's no reason not to upgrade and its the easiest thing to accomplish.

> If I upgrade my LXC container I am facing issues with apparmor cause the core18 kernel doesn't support it

apparmor issues are not relevant see https://github.com/nextcloud-snap/nextcloud-snap/wiki/Apparmor-profile-syslog

> Don't use LXC => Means I need to migrate my Nextcloud database and that I lose the flexibility of my container. Not ideal.

no way! LXD/LXC was the best choice by far for me personally, totally painless snapshotting containers, moving them from device to device, testing various setups see https://github.com/scubamuc/scubamuc.github.io 

recommend you upgrade your container to 24.04... simply run upgrades to 20.04, 22.04, 24.04 --> the snap will run perfectly since it has its "own" core and will keep using it until we're ready to migrate to core20 and up.... regardless which kernel the host uses.

**Comment by airlink75** [comment]
Hi @scubamuc ,

Thanks for your answer. I migrated my Nextcloud in 2024 from my old HP MicroServer to a new Proxmox server I setuped with always a LXC unprivileged container (currently running on Debian 12 but I planned to migrate on Debian 13 when I'll have time 😊).

It works like a charm and I am really satisfy. Just waiting like you for snap to be updated into core20 and up but it's not an issue for me now as I reached to run it on a supported os.

**Comment by pachulo** [comment]
Hi! We are now testing https://github.com/nextcloud-snap/nextcloud-snap/issues/3404

You can help us testing if everything works OK by upgrading to `latest/beta` with:
```
sudo snap refresh --beta nextcloud
```
